components/ActivityGridWrapper.tsx

```diff
@@ -0,0 +1,23 @@
+import { format } from 'date-fns';
+import ActivityGrid, { ActivityDay } from './ActivityGrid';
+
+export function makeGridArr(arr: { date: string }[]) {
+    let gridArr: ActivityDay[] = [];
+    for (let item of arr) {
+        const existingIndex = gridArr.findIndex(d => d.date === format(new Date(item.date), "yyyy-MM-dd"));
```

Comment:
this also seems inefficient, seems like you could do this in one pass rather than iterating for every element

Reply:
agree it sounds inefficient but idk how youd do it without iterating

Reply:
nvm i think i got it

Reply:
done